src/tests/Categories.test.tsx

```diff
@@ -1,35 +1,56 @@
 import { render, screen } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import { describe, expect, it } from 'vitest';
+import { beforeEach, describe, expect, it } from 'vitest';
 
 import { MemoryRouter } from 'react-router-dom';
 import { CategoryPage } from '../routes/CategoryPage';
 
 describe('Categories', () => {
-  it('should display buttons for each category', () => {
-    render(
-      <MemoryRouter>
-        <CategoryPage />
-      </MemoryRouter>
-    );
-
-    const genreButtons = screen.getAllByRole('button');
-    expect(genreButtons).toHaveLength(genreButtons.length);
+  beforeEach(() => {
+    render(<CategoryPage />, { wrapper: MemoryRouter });
+  });
+  const user = userEvent.setup();
+
+  it('should display dropdown', () => {
+    // expect dropdown
+    expect(screen.getByText('All')).toBeInTheDocument();
+
+    // expect correct icon to be displayed
+    expect(screen.getByTestId('dropdownIconDown')).toBeInTheDocument();
+    expect(screen.queryByTestId('dropdownIconUp')).not.toBeInTheDocument();
+  });
+
+  it('should open and display correct amout of categories when dropdown is clicked', async () => {
+    // expect and click dropdown
+    const dropdown = screen.getByText('All');
+    expect(dropdown).toBeInTheDocument();
+    await user.click(dropdown);
+
+    // expect correct icon to be displayed
+    expect(screen.getByTestId('dropdownIconUp')).toBeInTheDocument();
+    expect(screen.queryByTestId('dropdownIconDown')).not.toBeInTheDocument();
+
+    // except right amout of categories
+    expect(screen.getAllByRole('dropdownChoice')).toHaveLength(16);
   });
 
   it('should display right amout of movies with selected genre', async () => {
-    render(
-      <MemoryRouter>
-        <CategoryPage />
-      </MemoryRouter>
-    );
-    const user = userEvent.setup();
-
-    const actionGenreButton = screen.getByRole('button', { name: 'Action' });
-    expect(actionGenreButton).toBeInTheDocument();
-    await user.click(actionGenreButton);
-
-    const actionMovies = await screen.findAllByTestId('movie');
-    expect(actionMovies).toHaveLength(actionMovies.length);
+    // expect and click dropdown
+    const dropdown = screen.getByText('All');
+    expect(dropdown).toBeInTheDocument();
+    await user.click(dropdown);
+
+    // expect correct icon to be displayed
+    expect(screen.getByTestId('dropdownIconUp')).toBeInTheDocument();
+    expect(screen.queryByTestId('dropdownIconDown')).not.toBeInTheDocument();
+
+    // expect Drama category and select it
```

Comment:
Nitpicking here but surely it should be Adventure rather than Drama? 😅

Reply:
Good catch